turbopack/crates/turbopack-core/src/chunk/mod.rs

```diff
@@ -803,27 +746,35 @@ pub struct ChunkItems(pub Vec<ResolvedVc<Box<dyn ChunkItem>>>);
 
 #[turbo_tasks::value]
 pub struct AsyncModuleInfo {
-    pub referenced_async_modules: AutoSet<ResolvedVc<Box<dyn ChunkItem>>>,
+    pub referenced_async_modules: AutoSet<ResolvedVc<Box<dyn ChunkableModule>>>,
 }
 
 #[turbo_tasks::value_impl]
 impl AsyncModuleInfo {
     #[turbo_tasks::function]
-    pub async fn new(referenced_async_modules: Vec<Vc<Box<dyn ChunkItem>>>) -> Result<Vc<Self>> {
-        let resolved_modules = referenced_async_modules
-            .into_iter()
-            .map(|m| m.to_resolved())
-            .try_join()
-            .await?;
-
+    pub async fn new(
+        referenced_async_modules: Vec<ResolvedVc<Box<dyn ChunkableModule>>>,
+    ) -> Result<Vc<Self>> {
         Ok(Self {
-            referenced_async_modules: resolved_modules.into_iter().collect(),
+            referenced_async_modules: referenced_async_modules.into_iter().collect(),
         }
         .cell())
     }
 }
 
-pub type ChunkItemWithAsyncModuleInfo = (Vc<Box<dyn ChunkItem>>, Option<Vc<AsyncModuleInfo>>);
+#[turbo_tasks::value]
+pub enum ChunkItemTy {
+    /// The ChunkItem should be included as content in the chunk.
+    Included,
+    /// The ChunkItem should be used to trace references but should not included in the chunk.
+    Passthrough,
+}
+
+pub type ChunkItemWithAsyncModuleInfo = (
+    Vc<ChunkItemTy>,
```

Comment:
You want to avoid using `Vc` for simple Copy values: (except for return values where it has to be a Vc)

```suggestion
    ChunkItemTy,
```